src/gui/transferlistfilters/trackersfilterwidget.cpp

```diff
@@ -335,7 +335,13 @@ void TrackersFilterWidget::handleTrackerEntriesUpdated(const BitTorrent::Torrent
                 errored.remove(trackerEntry.url);
             }
 
-            if (trackerEntry.message.isEmpty())
+            if (std::all_of(trackerEntry.stats.cbegin(), trackerEntry.stats.cend(), [](const auto &endpoint) {
+                  return std::all_of(endpoint.cbegin(), endpoint.cend(),
+                                     [](const BitTorrent::TrackerEntry::EndpointStats &protocolStats) {
+                                       return protocolStats.status != BitTorrent::TrackerEntry::Working ||
+                                              protocolStats.message.isEmpty();
+                                     });
+                }))
```

Comment:
It looks like it doesn't match the previous condition (because of `protocolStats.status != BitTorrent::TrackerEntry::Working`).

Reply:
Also please don't put all this code in `if` condition directly but assign to some variable.

Reply:
```suggestion
            const bool hasNoWarningMessages = std::all_of(trackerEntry.stats.cbegin(), trackerEntry.stats.cend(), [](const auto &endpoint)
            {
                  return std::all_of(endpoint.cbegin(), endpoint.cend()
                          , [](const BitTorrent::TrackerEntry::EndpointStats &protocolStats)
                  {
                      return protocolStats.message.isEmpty() || (protocolStats.status != BitTorrent::TrackerEntry::Working);
                  });
            });
            if (hasNoWarningMessages)
```